server/channels/app/desktop_login.go

```diff
@@ -0,0 +1,81 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+package app
+
+import (
+	"net/http"
+
+	"github.com/mattermost/mattermost/server/public/model"
+)
+
+func (a *App) SaveClientDesktopToken(token string, createdAt int64) *model.AppError {
+	// Create token in the database
+	err := a.Srv().Store().DesktopTokens().Insert(token, createdAt, nil)
+	if err != nil {
+		return model.NewAppError("SaveClientDesktopToken", "app.desktop_token.create.error", nil, err.Error(), http.StatusInternalServerError)
+	}
+
+	return nil
+}
+
+func (a *App) AuthenticateClientDesktopToken(token string, expiryTime int64, user *model.User) *model.AppError {
+	// Throw an error if the token is expired
+	err := a.Srv().Store().DesktopTokens().SetUserId(token, expiryTime, user.Id)
+	if err != nil {
+		// Delete the token if it is expired
```

Comment:
Any reason we are spawning so many goroutines unnecessarily in these methods? There are lots of pitfalls with unbounded goroutine creation, especially ones that make DB queries: https://developers.mattermost.com/contribute/more-info/server/style-guide/#default-to-sync-instead-of-async.

> Do not create one-off goroutines without knowing when/how they exit.

Unless there is a strong reason for it, the default behavior should be to _not_ use a goroutine.

Reply:
So I did this as a `defer` originally because I didn't want to lock the database by deleting a row synchronously. If this is not a major concern, I will remove the goroutine and go back to doing it synchronously.

Reply:
The locking will anyways happen regardless of you use a goroutine or not. It's just that now the load on the DB won't be unbounded.